app/controllers/declaration/PreviousDocumentsController.scala

```diff
@@ -19,79 +19,55 @@ package controllers.declaration
-import forms.declaration.officeOfExit.AllowedUKOfficeOfExitAnswers.{no, yes}
-import forms.declaration.officeOfExit.OfficeOfExitOutsideUK
 import forms.declaration.{Document, PreviousDocumentsData}
-import handlers.ErrorHandler
 import javax.inject.Inject
-import models.DeclarationType.DeclarationType
 import models.requests.JourneyRequest
-import models.{DeclarationType, ExportsDeclaration, Mode}
+import models.{ExportsDeclaration, Mode}
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import services.cache.ExportsCacheService
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
-import views.html.declaration.previous_documents
+import views.html.declaration.previousDocuments.previous_documents
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class PreviousDocumentsController @Inject()(
   authenticate: AuthAction,
   journeyType: JourneyAction,
   navigator: Navigator,
-  errorHandler: ErrorHandler,
   mcc: MessagesControllerComponents,
   previousDocumentsPage: previous_documents,
   override val exportsCacheService: ExportsCacheService
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable with SubmissionErrors {
 
   def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    val frm = form().withSubmissionErrors()
-    request.cacheModel.previousDocuments match {
-      case Some(data) => Ok(previousDocumentsPage(mode, frm, data.documents))
-      case _          => Ok(previousDocumentsPage(mode, frm, Seq.empty))
-    }
+    Ok(previousDocumentsPage(mode, form()))
   }
 
   def savePreviousDocuments(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    import MultipleItemsHelper._
-
-    val boundForm = form().bindFromRequest()
-    val actionTypeOpt = FormAction.bindFromRequest()
+    val boundForm = if (isFirstPreviousDocument) Document.treatLikeOptional(form().bindFromRequest()) else form().bindFromRequest()
 
     val cache = request.cacheModel.previousDocuments.getOrElse(PreviousDocumentsData(Seq.empty))
 
-    actionTypeOpt match {
-      case SaveAndContinue | SaveAndReturn =>
-        saveAndContinue(boundForm, cache.documents, isScreenMandatory, maxAmountOfItems).fold(
-          formWithErrors => Future.successful(BadRequest(previousDocumentsPage(mode, formWithErrors, cache.documents))),
-          updatedCache =>
-            updateCache(PreviousDocumentsData(updatedCache))
-              .map(_ => navigator.continueTo(mode, controllers.declaration.routes.ItemsSummaryController.displayPage))
-        )
-
-      case Add =>
-        add(boundForm, cache.documents, PreviousDocumentsData.maxAmountOfItems).fold(
-          formWithErrors => Future.successful(BadRequest(previousDocumentsPage(mode, formWithErrors, cache.documents))),
+    if (boundForm.value.isEmpty && boundForm.errors.isEmpty)
```

Comment:
Nit: Just wondering if it is possible to combine these two `if` statements into one? I don't know enough about business requirements to tell.

Reply:
Not sure about this, this is based on the play framework, form will be empty if has errors or it's empty, so I'm not sure if this can be shorter